tools/realsense-viewer/realsense-viewer.cpp

```diff
@@ -273,6 +276,216 @@ bool refresh_devices(std::mutex& m,
 
 int main(int argc, const char** argv) try
 {
+
+
+////// Test Auto Update //////////
+#ifdef ENABLE_RS_AUTO_UPDATER
+//#define AUTO_UPDATER_UNIT_TEST  
+
+#ifdef AUTO_UPDATER_UNIT_TEST
+
+    // Set 3 callback functions for testing - SHOULD BE REMOVED!!
+
+    // Main Success Scenario callback
+    std::function<callback_result(uint64_t dl_current_bytes, uint64_t dl_total_bytes)> mss_process_cb =
+        [](uint64_t dl_current_bytes, uint64_t dl_total_bytes) -> callback_result {
+        std::cout << "DOWN:" << dl_current_bytes << " of " << dl_total_bytes << std::endl;
+        return callback_result::CONTINUE_DOWNLOAD;
+    };
+
+    std::function<callback_result(uint64_t dl_current_bytes, uint64_t dl_total_bytes)> stop_process_cb =
+        [](uint64_t dl_current_bytes, uint64_t dl_total_bytes) -> callback_result {
+        std::cout << "DOWN:" << dl_current_bytes << " of " << dl_total_bytes << std::endl;
+        return callback_result::STOP_DOWNLOAD;
+    };
+
+    std::function<callback_result(uint64_t dl_current_bytes, uint64_t dl_total_bytes)> empty_process_cb;
+
+
+    //////////////// TEST 1 START ////////////////
+    // Download from corrupted format URL test  //
+    // Expect Download fail                     //
+    //////////////////////////////////////////////
+    {
+        versions_db_manager up_handler("Fake_URL...json");
+        versions_db_manager::version ver;
+        bool res(false);
+        try {
+            res = up_handler.query_versions("Intel RealSense L515", versions_db_manager::LIBREALSENSE, versions_db_manager::RECOMMENDED, ver);
+        }
+        catch (...)
+        {
+            assert(false);
+        }
+        assert(false == res);
+    }
+
+    /////////// TEST 2 START /////////////
+    // Download from invalid URL test   //
+    // Expect Download fail             //
+    //////////////////////////////////////
+    {
+        versions_db_manager up_handler("Fake_URL.json");
+        versions_db_manager::version ver;
+        bool res(false);
+        try {
+            res = up_handler.query_versions("Intel RealSense L515", versions_db_manager::LIBREALSENSE, versions_db_manager::RECOMMENDED, ver);
+        }
+        catch (...)
+        {
+            assert(false);
+        }
+        assert(false == res);
+    }
+
+    /////////////// TEST 3 START ///////////////////
+    // Download from s3 - not a version json      //
+    // Expect Download OK - Parse Fail            //
+    ////////////////////////////////////////////////
+    {
+        versions_db_manager up_handler("http://realsense-hw-public.s3.amazonaws.com/rs-tests/post_processing_tests_2018_ww18/1551257880762.0.Input.csv");
```

Comment:
Gather hard-coded values in the beginning of cpp

Reply:
This section will be removed before the main PR.
it is a temporary section that allow me to run regression tests during development and fixes .
When it will be a formal unit test I will remember this comment.
For now it is more comfortable for me to leave it as it is.

Reply:
Ok. Just put a //TODO comment in the code